server/models/Transaction.js

```diff
@@ -506,5 +510,94 @@ export default (Sequelize, DataTypes) => {
+        type: 'DEBIT',
+      },
+    });
+    transaction.debitId = debitTransaction.id;
+    if (transaction.HostCollectiveId) {
+      const hostCollective = await models.Collective.findById(transaction.HostCollectiveId);
+      transaction.hostCollectiveSlug = hostCollective.slug;
+      const paymentMethod = await models.PaymentMethod.findById(transaction.PaymentMethodId);
+      transaction.paymentMethodService = paymentMethod.service;
+      transaction.paymentMethodType = paymentMethod.type;
+      if (paymentMethod.CollectiveId) {
+        transaction.paymentMethodCollectiveId = paymentMethod.CollectiveId;
+        const pmCollective = await models.Collective.findById(paymentMethod.CollectiveId);
+        transaction.paymentMethodCollectiveSlug = pmCollective.slug;
+      }
+    }
+    if (transaction.OrderId) {
+      const order = await models.Order.findById(transaction.OrderId);
+      if (order.FromCollectiveId) {
+        transaction.orderFromCollectiveId = order.FromCollectiveId;
+        const orderFromCollective = await models.Collective.findById(order.FromCollectiveId);
+        transaction.orderFromCollectiveSlug = orderFromCollective.slug;
+      }
+      if (order.PaymentMethodId) {
+        const orderPaymentMethod = await models.PaymentMethod.findById(order.PaymentMethodId);
+        transaction.orderPaymentMethodService = orderPaymentMethod.service;
+        transaction.orderPaymentMethodType = orderPaymentMethod.type;
+        if (orderPaymentMethod.CollectiveId) {
+          transaction.orderPaymentMethodCollectiveId = orderPaymentMethod.CollectiveId;
+          const orderPaymentMethodCollective = await models.Collective.findById(orderPaymentMethod.CollectiveId);
+          transaction.orderPaymentMethodCollectiveSlug = orderPaymentMethodCollective.slug;
+        }
+      }
+    }
+    if (transaction.ExpenseId) {
+      const expense = await models.Expense.findById(transaction.ExpenseId);
+      transaction.expensePayoutMethod = expense.payoutMethod;
+      if (expense.UserId) {
+        transaction.expenseUserId = expense.UserId;
+        const expenseUser = await models.User.findById(expense.UserId);
+        transaction.expenseUserPaypalEmail = expenseUser.paypalEmail;
+        if (expenseUser.CollectiveId) {
+          const expenseUserCollective = await models.Collective.findById(expenseUser.CollectiveId);
+          transaction.expenseUserCollectiveSlug = expenseUserCollective.slug;
+        }
+      }
+      if (expense.CollectiveId) {
+        const expenseCollective = await models.Collective.findById(expense.CollectiveId);
+        transaction.expenseCollectiveId = expense.CollectiveId;
+        transaction.expenseCollectiveSlug = expenseCollective.slug;
+      }
+    }
+    return transaction;
+  };
+
+  Transaction.postTransactionToLedger = async transaction => {
+    if (transaction.deletedAt || transaction.type !== 'CREDIT') {
+      return Promise.resolve();
+    }
+    try {
```

Comment:
Can we move that, maybe not the `postTransactionToLedger` itself, but what's inside into `lib/ledger`? Also add `getTransactionWithNestedProperties`.

Reply:
Sure, makes sense. I've created the `lib/ledger` on the other branch. Would you mind if we then merge the [*read*](https://github.com/opencollective/opencollective-api/pull/1578) branch before so I can rebase from master here and do this change?